From: Shaohua Li <shaohua.li@intel.com>
Date: Thu, 3 Dec 2009 12:58:05 +0100
Subject: [PATCH] cfq-iosched: no dispatch limit for single queue
Git-commit: 474b18ccc264c472abeec50f48469b6477202699
References: FATE#311054
Patch-Mainline: 2.6.33

Since commit 2f5cb7381b737e24c8046fd4aeab571fb71315f5, each queue can send
up to 4 * 4 requests if only one queue exists. I wonder why we have such limit.
Device supports tag can send more requests. For example, AHCI can send 31
requests. Test (direct aio randread) shows the limits reduce about 4% disk
thoughput.
On the other hand, since we send one request one time, if other queue
pop when current is sending more than cfq_quantum requests, current queue will
stop send requests soon after one request, so sounds there is no big latency.

Signed-off-by: Shaohua Li <shaohua.li@intel.com>
Signed-off-by: Jens Axboe <jens.axboe@oracle.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 block/cfq-iosched.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/block/cfq-iosched.c b/block/cfq-iosched.c
index ed7f2c5..1161dd8 100644
--- a/block/cfq-iosched.c
+++ b/block/cfq-iosched.c
@@ -1620,9 +1620,9 @@ static bool cfq_may_dispatch(struct cfq_data *cfqd, struct cfq_queue *cfqq)
 			return false;
 
 		/*
-		 * Sole queue user, allow bigger slice
+		 * Sole queue user, no limit
 		 */
-		max_dispatch *= 4;
+		max_dispatch = -1;
 	}
 
 	/*
-- 
1.6.0.2

